### PR TITLE
chore: add initial configuration for `mergeable`

### PR DESCRIPTION
**Pull request by tenshiAMD**
This includes a lot of necessary checks which mostly follow semantic release convention.

### CODE REVIEW

**Comment by tenshiAMD** [comment]
This is for https://github.com/KalleHallden/exer_log/issues/109.

File: .github/mergeable.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          regex: \[ \]
+          message: There are incomplete TODO task(s) unchecked.
+      - do: approvals
+        min:
+          count: 1
+        or:
+          - required:
+              reviewers: [ jusx ]
+          - required:
+              reviewers: [ shine2lay ]
```
Comment (jorre127) [inline .github/mergeable.yml]: Not familiar with mergeable so don't know if I'm reading this correctly. But does this mean that a pullrequest has to be reviewed by jusx or shine2lay? if so who are they.

Comment (tenshiAMD) [reply .github/mergeable.yml]: @jorre127 My bad. I forgot to remove those. Anyways, I updated this PR. Please check again. Thank you.

File: .github/mergeable.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    validate:
+      - do: approvals
+        min:
+          count: 1
+  - when: pull_request.*, pull_request_review.*
+    validate:
+      - do: dependent
+        changed:
+          file: 'pubspec.yaml'
+          required: ['pubspec.lock']
```
Comment (momshaddinury) [inline .github/mergeable.yml:50]: Should we make pubspec.lock required? Aren't generated files supposed to be ignored?

Comment (tenshiAMD) [reply .github/mergeable.yml:50]: @dinurymomshad I think you misunderstand it. Let me clarify this one, so if there are changes in `pubspec.yaml` the PR should check and must include changes in `pubspec.lock` as a dependency.

**Comment by momshaddinury** [comment]
@KalleHallden this looks good. We can merge this.

**Comment by tenshiAMD** [comment]
@KalleHallden Please follow this [installation guide](https://mergeable.readthedocs.io/en/latest/usage.html) to make this work. Thank you.

**Pull request merged**